sdks/java/core/src/test/java/org/apache/beam/sdk/testing/ExpectedLogsTest.java

```diff
@@ -146,6 +152,11 @@ public Void call() throws Exception {
     }
   }
 
+  @Test
+  public void testLogsCleared() {
```

Comment:
Look at RestoreDataflowLoggingMDCTest to see how to create a test rule and validate the before/after state.